src/redis-cli.c

```diff
@@ -9820,6 +10013,571 @@ void testHintSuite(char *filename) {
+            exit(1);
+        }
+        /* Keep track of the biggest size for this type */
+        type->biggest = size;
+    }
+}
+
+static void keyStats(long long memkeys_samples, unsigned long long cursor, unsigned long top_sizes_limit) {
+    unsigned long long sampled = 0, total_keys, total_size = 0, it = 0, scan_loops = 0;
+    unsigned long long *memkeys_sizes = NULL, *bigkeys_sizes = NULL;
+    redisReply *reply, *keys;
+    unsigned int array_size = 0, i;
+    typeinfo **memkeys_types = NULL, **bigkeys_types = NULL;
+    list *top_sizes;
+
+    if (cursor != 0) {
+        it = cursor;
+    }
+
+    if ((top_sizes = listCreate()) == NULL){
+	    fprintf(stderr, "top_sizes list creation failed.\n");
+	    exit(1);
+    }
+    top_sizes->free = zfree;
+
+    dict *memkeys_types_dict = dictCreate(&typeinfoDictType);
+    typeinfo_add(memkeys_types_dict, "string", &type_string);
+    typeinfo_add(memkeys_types_dict, "list", &type_list);
+    typeinfo_add(memkeys_types_dict, "set", &type_set);
+    typeinfo_add(memkeys_types_dict, "hash", &type_hash);
+    typeinfo_add(memkeys_types_dict, "zset", &type_zset);
+    typeinfo_add(memkeys_types_dict, "stream", &type_stream);
+
+    /* We could use only one typeinfo dictionary if we add new fields to save
+     * both memkey and bigkey info. Not sure it would make sense in findBigKeys(). */
+    dict *bigkeys_types_dict = dictCreate(&typeinfoDictType);
+    typeinfo_add(bigkeys_types_dict, "string", &type_string);
+    typeinfo_add(bigkeys_types_dict, "list", &type_list);
+    typeinfo_add(bigkeys_types_dict, "set", &type_set);
+    typeinfo_add(bigkeys_types_dict, "hash", &type_hash);
+    typeinfo_add(bigkeys_types_dict, "zset", &type_zset);
+    typeinfo_add(bigkeys_types_dict, "stream", &type_stream);
+
+    size_dist key_length_dist;
+    size_dist_entry distribution[] = {
+        {1<<5, 0},                 /*  32 B  (sds)                                            */
+        {1<<8, 0},                 /* 256 B  (sds)                                            */
+        {1<<16, 0},                /*  64 KB (sds and Redis Enterprise key name max length)   */
+        {1024*1024, 0},            /*   1 MB                                                  */
+        {16*1024*1024, 0},         /*  16 MB                                                  */
+        {128*1024*1024, 0},        /* 128 MB                                                  */
+        {512*1024*1024, 0},        /* 512 MB (max String size)                                */
+        {0, 0},                    /* Sizes above the last entry                              */
+    };
+    sizeDistInit(&key_length_dist, distribution);
+
+    struct hdr_histogram* keysize_histogram;
+    /* Record max of 1TB for a key size should cover all keys.
+     * significant_figures == 4 (0.01% precision on key size)  */
+    if (hdr_init(1, INT64_C(1ULL*1024*1024*1024*1024), 4, &keysize_histogram)) {
```

Comment:
`1ULL` and UINT64_C are the same.
```suggestion
    if (hdr_init(1, 1ULL*1024*1024*1024*1024, 4, &keysize_histogram)) {
```